packages/ui/src/SplitPage.native.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {Children, useCallback, useEffect, useState} from "react";
```

Comment:
One thing where splitting this native vs web like this will bite us is when we try to support apps on tablets. Those would want to get the other experience, but would get a giant phone experience right now.

Not needed for this PR, but something we'll likely want to tackle in the medium term.